drivers/ethernet/eth_nxp_imx_netc_psi.c

```diff
@@ -0,0 +1,183 @@
+#include <zephyr/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/phy.h>
+#include <ethernet/eth_stats.h>
+
+#include "eth.h"
+#include "eth_nxp_imx_netc_priv.h"
+
+static void netc_eth_phylink_callback(const struct device *pdev, struct phy_link_state *state,
+				      void *user_data)
+{
+	const struct device *dev = (struct device *)user_data;
+	struct netc_eth_data *data = dev->data;
+	status_t result;
+
+	ARG_UNUSED(pdev);
+
+	if (state->is_up) {
+		LOG_DBG("Link up");
+		result = EP_Up(&data->handle, PHY_TO_NETC_SPEED(state->speed),
+			       PHY_TO_NETC_DUPLEX_MODE(state->speed));
+		if (result != kStatus_Success) {
+			LOG_ERR("Failed to set MAC up");
+		}
+		net_eth_carrier_on(data->iface);
+	} else {
+		LOG_DBG("Link down");
+		result = EP_Down(&data->handle);
+		if (result != kStatus_Success) {
+			LOG_ERR("Failed to set MAC down");
+		}
+		net_eth_carrier_off(data->iface);
+	}
+}
+
+static void netc_eth_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct netc_eth_data *data = dev->data;
+	const struct netc_eth_config *cfg = dev->config;
+	status_t result;
+
+	/*
+	 * For VLAN, this value is only used to get the correct L2 driver.
+	 * The iface pointer in context should contain the main interface
+	 * if the VLANs are enabled.
+	 */
+	if (data->iface == NULL) {
+		data->iface = iface;
+	}
+
+	/* Set MAC address */
+	result = EP_SetPrimaryMacAddr(&data->handle, (uint8_t *)data->mac_addr);
+	if (result != kStatus_Success) {
+		LOG_ERR("Failed to set MAC address");
+	}
+
+	net_if_set_link_addr(iface, data->mac_addr, sizeof(data->mac_addr), NET_LINK_ETHERNET);
+
+	LOG_INF("SI%d MAC: %02x:%02x:%02x:%02x:%02x:%02x", cfg->si_idx, data->mac_addr[0],
+		data->mac_addr[1], data->mac_addr[2], data->mac_addr[3], data->mac_addr[4],
+		data->mac_addr[5]);
+
+	ethernet_init(iface);
+
+	/*
+	 * PSI controls the PHY. If PHY is configured either as fixed
+	 * link or autoneg, the callback is executed at least once
+	 * immediately after setting it.
+	 */
+	if (!device_is_ready(cfg->phy_dev)) {
+		LOG_ERR("PHY device (%p) is not ready, cannot init iface", cfg->phy_dev);
+		return;
+	}
+	phy_link_callback_set(cfg->phy_dev, &netc_eth_phylink_callback, (void *)dev);
+
+	/* Do not start the interface until PHY link is up */
+	net_if_carrier_off(iface);
+}
+
+static int netc_eth_init(const struct device *dev)
+{
+	const struct netc_eth_config *cfg = dev->config;
+	int err;
+
+	err = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	return netc_eth_init_common(dev);
+}
+
+static const struct ethernet_api netc_eth_api = {.iface_api.init = netc_eth_iface_init,
+						 .get_capabilities = netc_eth_get_capabilities,
+						 .set_config = netc_eth_set_config,
+						 .send = netc_eth_tx};
+
+#define PHY_NODE(n) DT_INST_PHANDLE(n, phy_handle)
```

Comment:
nit, this definition is not needed, just use DT_INST_PHANDLE inline

Reply:
Okay, thanks.